sql-parser/src/main/java/io/crate/sql/tree/SubqueryExpression.java

```diff
@@ -25,6 +25,8 @@
     extends Expression {
     private final Query query;
 
+    private boolean isArrayExpression;
```

Comment:
The `SubqueryExpression` has to result in an array expression in an `ANY` operation. Otherwise it is not valid.

The following are interchangeable:

`select id where id = ANY([1])` <-- array argument
`select id where id = ANY(Select 1)` <-- array argument